webapi/src/main/scala/org/knora/webapi/core/AppServer.scala

```diff
@@ -97,19 +95,15 @@ final case class AppServer(
   private def checkIIIFService(requiresIIIFService: Boolean): ZIO[Any, Nothing, Unit] =
     for {
       _ <- state.set(AppState.WaitingForIIIFService)
-      _ <- if (requiresIIIFService)
-             iiifs
-               .getStatus()
-               .flatMap(status =>
-                 status match {
-                   case IIIFServiceStatusOK =>
-                     ZIO.logInfo("IIIF service running")
-                   case IIIFServiceStatusNOK =>
-                     ZIO.logError("IIIF service not running") *> ZIO.die(new Exception("IIIF service not running"))
-                 }
-               )
-           else
-             ZIO.unit
+      _ <- iiifs
+             .getStatus()
+             .flatMap {
+               case IIIFServiceStatusOK =>
+                 ZIO.logInfo("IIIF service running")
+               case IIIFServiceStatusNOK =>
+                 ZIO.logError("IIIF service not running") *> ZIO.die(new Exception("IIIF service not running"))
+             }
+             .when(requiresIIIFService)
       _ <- state.set(AppState.IIIFServiceReady)
```

Comment:
I'm just wondering if it's actually reasonable to set the AppState to IIIF stuff, if we don't require the IIIF service. So maybe this method could be rewritten to:

```scala
private def checkIIIFService(requiresIIIFService: Boolean): ZIO[Any, Nothing, Unit] =
  if (requiresIIIFService) {
    for {
      _      <- state.set(AppState.WaitingForIIIFService)
      status <- iiifs.getStatus()
      _      <- status match {
        case CacheServiceStatusNOK =>
          ZIO.logError("Cache service not running.") *> 
            ZIO.die(new Exception("Cache service not running."))
        case CacheServiceStatusOK =>
          ZIO.unit
      }
      _      <- state.set(AppState.IIIFServiceReady)
    } yield ()
  } else ZIO.unit
```

optionally, you could also use ZIO.when to test the bool condition. And even ZIO.whenCase to check the status

Reply:
yes, good point. With ZIO I'm actually not sure if we need these states in such a detail at all, or if something waiting for startup is not good enough. Now that we have logging that works, we should quickly figure out where the startup is stuck. So I will merge it as is, and we can create a follow-up issue after we discuss it.